### PR TITLE
feat: adds support for audience in client_options

### PR DESCRIPTION
**Pull request by atulep**
Adds support for `audience` in client_options.

### CODE REVIEW

**Comment by atulep** [comment]
@viacheslav-rostovtsev FYI. This is needed to make `audience` configurable in the downstream GAPIC client.

**Comment by atulep** [comment]
@viacheslav-rostovtsev What can be a good documentation for `audience`?

File: google/api_core/client_options.py [hunk 1]
```diff
@@ -70,6 +70,7 @@ class ClientOptions(object):
         scopes (Optional[Sequence[str]]): OAuth access token override scopes.
         api_key (Optional[str]): Google API key. ``credentials_file`` and
             ``api_key`` are mutually exclusive.
+        audience (Optional[str]): The desired API audience.
```
Comment (parthea) [inline google/api_core/client_options.py:73]: Please could you include an example string to show the correct format?

**Review by parthea — APPROVED** [review]
Added one observation

**Comment by viacheslav-rostovtsev** [comment]
- let's call the parameter `api_audience` and not just `audience` because it looks like in the TPC work we might get other type of audience as well.
- The documentation should be something like: 'The intended audience for the API calls to the service that will be set when using certain 3rd party authentication flows. Audience is typically a resource identifier. If not set, the service endpoint value will be used as a default.'
- Good examples of API audience are google service endpoints like `https://language.googleapis.com`

**Review by viacheslav-rostovtsev — CHANGES_REQUESTED** [review]
Let's change this to `api_audience`.

File: google/api_core/client_options.py (hunk 1 above)
Comment (atulep) [reply google/api_core/client_options.py]: Done

**Comment by atulep** [comment]
@viacheslav-rostovtsev PTAL again

**Review by viacheslav-rostovtsev — APPROVED** [review]
LGTM

File: google/api_core/client_options.py [hunk 2]
```diff
@@ -70,6 +70,11 @@ class ClientOptions(object):
         scopes (Optional[Sequence[str]]): OAuth access token override scopes.
         api_key (Optional[str]): Google API key. ``credentials_file`` and
             ``api_key`` are mutually exclusive.
+        api_audience (Optional[str]): The intended audience for the API calls
+            to the service that will be set when using certain 3rd party
+            authentication flows. Audience is typically a resource identifier.
+            If not set, the service endpoint value will be used as a default.
+            An example of a valid ``api_audience`` is: "https://language.googleapis.com".
```
Comment (nicain) [inline google/api_core/client_options.py:77]: language.googleapis.com is used as an example here.

**Pull request merged**